provisioning/baremetal_pod.go

```diff
@@ -314,6 +371,9 @@ func createInitContainerStaticIpSet(images *Images, config *metal3iov1alpha1.Pro
 		Command:         []string{"/set-static-ip"},
 		ImagePullPolicy: "IfNotPresent",
 		SecurityContext: &corev1.SecurityContext{
+			ReadOnlyRootFilesystem: ptr.To(true),
```

Comment:
Privileged container is needed for it to create a symlink for the host path volume for shared image cache. I have removed the `privileged: true` from the pod and individually added it to containers when needed.